piqueserver/config.py

```diff
@@ -0,0 +1,86 @@
+import toml
+from pprint import pprint
+
+class ConfigException(Exception):
+    pass
+
+
+class ConfigEngine():
+    '''
+    configuration engine that handles loading and saving config,
+    and all the things
+    '''
+    def __init__(self):
+        self.raw_config = {}
+        self.style = 'TOML'
+
+    def load_config(self, config_file, style='TOML'):
+        # TODO: could also have this merge into current config if available
+        self.style = style
+
+        if style == 'TOML':
+            self.raw_config = toml.load(open(config_file))
+        else:
+            raise Exception()
+
+    def dump_config(self, out_file=None, style='TOML'):
+        if out_file is None:
+            pprint(self.raw_config)
+        # TODO
+        pass
+
+    def get(self, name, default):
+        # TODO: handle nested config values
+        # XXX: should requesting a config value that is a dictionary object
+        #       return the whole dictionary, or fail?
+        return self.raw_config.get(name, default)
+
+    def set(self, name, value):
+        self.raw_config[name] = value
+
+    def __call__(self, name, cast=None, default=None, validate=None):
```

Comment:
I think calling this `option()` or `setting()` or something would make more sense as you are declaring a config option, not a config. This would also make other types of settings, such as files or settings a lot more intuitive.